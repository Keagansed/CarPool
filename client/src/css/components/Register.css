#inputFirstName, 
#inputLastName,
#inputID,
#inputEmail
{
    background-color: rgba(0,0,0,0) !important;
    color: white;
    border: 0.05rem solid #6FEDB7;
}

/* Change Autocomplete styles in Chrome*/
input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus {
    -webkit-text-fill-color: white;
    transition: background-color 5000s ease-in-out 0s;
} 