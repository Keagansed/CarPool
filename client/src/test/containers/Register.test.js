import React from 'react';
import Register from '../../js/containers/RegisterPage';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

class MockStore {
    token = null; 
    loggedIn = false; 
    registered = false; 
    lemail = ''; 
    lpassword = '';
    sFName = '';
    sLName = '';
    semail = '';
    sPass1 = '';
    sPass2 = '';
    sId = '';

    setToken = (token) => {
        this.token = token;
    };

    setLoggedIn = (success) => {
        this.loggedIn = success;
        this.lemail = '';
        this.lpassword = '';
    };

    setRegistered = (val) => {
        this.registered = val;
    };

    signUp = () => {
        
        if(this.sPass1 === this.sPass2) {
            this.setRegistered(true);
        }

    };

    authenticate = () => {
        this.setRegistered(false);
        this.setToken("testtoken123");
        this.setLoggedIn(true);
    };

    logout = () => {
        this.setLoggedIn(false);
        this.setToken('');
    };

}

describe('Register Component', () => {
    let mockStore, container, instance;
    
    beforeAll(() => {
        mockStore = new MockStore();
        container = shallow(<Register store={mockStore} />);
        instance = container.instance();
    });
    
    it('captures snapshot', () => {
        const renderedValue = renderer.create(<Register store={mockStore} />).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
  
    it('renders correctly', () => {
        expect(container.length).toEqual(1);
    });

    it('updates sign up first name correctly', () => {
        const event = {target: {value: "test"}};
        expect(instance.state.fName).toEqual('');
        instance.updateSignUpfNameValue(event);
        expect(instance.state.fName).toEqual("test");
    });

    it('updates sign up last name correctly', () => {
        const event = {target: {value: "tester"}};
        expect(instance.state.lName).toEqual('');
        instance.updateSignUplNameValue(event);
        expect(instance.state.lName).toEqual("tester");
    });

    it('updates sign up email correctly', () => {
        const event = {target: {value: "test@email.com"}};
        expect(instance.state.email).toEqual('');
        instance.updateSignUpEmailValue(event);
        expect(instance.state.email).toEqual("test@email.com");
    });

    it('updates sign up ID correctly', () => {
        const event = {target: {value: "9311179351087"}};
        expect(instance.state.idNum).toEqual('');
        instance.updateSignUpIDValue(event);
        expect(instance.state.idNum).toEqual("9311179351087");
    });

    it('updates first sign up password correctly', () => {
        const event = {target: {value: "1234"}};
        expect(instance.state.password1).toEqual('');
        instance.updateSignUpPasswordValue1(event);
        expect(instance.state.password1).toEqual("1234");
    });

    it('updates second sign up password correctly', () => {
        const event = {target: {value: "1234"}};
        expect(instance.state.password2).toEqual('');
        instance.updateSignUpPasswordValue2(event);
        expect(instance.state.password2).toEqual("1234");
    });

    it('handles sign up correctly', () => {
        const event = {preventDefault: function() {}};
        expect(instance.props.store.registered).toEqual(false);
        instance.handleSignup(event);
        expect(instance.props.store.registered).toEqual(true);
    });

    it('checks state variables correctly before submitting', () => {
        expect(instance.canBeSubmitted()).toEqual(true);
    });

    it('checks invalid state variables correctly before submitting', () => {
        instance.state.email = "notavalidemailaddress";
        expect(instance.canBeSubmitted()).toEqual(false);
    });
});