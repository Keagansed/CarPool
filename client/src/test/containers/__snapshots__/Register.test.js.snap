// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component captures snapshot 1`] = `
<div
  className="vertical-center bg-purple"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <img
        alt="carpool_logo"
        className="img-fluid d-block mx-auto mbottom-1rem"
        id="logo-256"
        src="logo.png"
      />
    </div>
    <form>
      <div
        className="row"
      >
        <input
          className=" form-control mx-auto width-15rem brad-2rem mbottom-1rem"
          id="inputFirstName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <input
          className=" form-control mx-auto width-15rem brad-2rem mbottom-1rem"
          id="inputLastName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Last Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <input
          className=" form-control mx-auto width-15rem brad-2rem mbottom-1rem"
          id="inputID"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="ID Number"
          type="text"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <input
          className=" form-control mx-auto width-15rem brad-2rem mbottom-1rem"
          id="inputEmail"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <input
          className=" form-control mx-auto width-15rem brad-2rem mbottom-1rem"
          id="inputPassword"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <input
          className=" form-control mx-auto width-15rem brad-2rem mbottom-1rem"
          id="inputConfirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Confirm Password"
          type="password"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <button
          className="btn btn-primary mx-auto width-15rem brad-2rem mbottom-1rem bg-aqua txt-purple fw-bold"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</div>
`;
